### PR TITLE
fix: Replace sentinel value calculation with something correct.

### PR DESCRIPTION
**Pull request by barbarj**
I was mistaken that `num_vals` is the number of distinct non-nil terms in a column. It is instead the number of "values" in that column, non-distinct. I've modified the agg missing sentinel tests to to include docs with duplicate terms to exhibit this behavior.

The fix is then to use `accessor.max_value()` as before, in combination with `accessor.values.num_vals()` to handle the 0-values case.

### CODE REVIEW

File: src/aggregation/accessor_helpers.rs [hunk 1]
```diff
@@ -16,15 +16,22 @@ use crate::index::SegmentReader;
...
         Key::Str(_) | Key::F64(_) | Key::U64(_) | Key::I64(_) if column_type == ColumnType::Str => {
-            // For strings, we use the max term ordinal + 1, which happens to always be the same as
-            // the number of values in the column
-            Some(column_val_count as u64)
+            // For strings, we use the max term ordinal + 1. If the column has >0 non-nil values,
+            // then we can get the correct max term ordinal from the max column value. However, when
+            // the column has no non-nil values, the max column value defaults to 0, which is
+            // actually the sentinel value we want to use, so we must specify that directly
+            if column_value_count > 0 {
+                Some(column_max_value + 1)
```
Comment (stuhood) [inline src/aggregation/accessor_helpers.rs]: This should be checked math probably: unlikely to encounter overflow for a string dictionary, but it's much more likely to encounter when dealing with a `max`.

**Review by stuhood — APPROVED** [review]
If you're sure the change to the test is enough to cover this, then #shipit.

File: src/aggregation/accessor_helpers.rs (hunk 1 above)
Comment (barbarj) [reply src/aggregation/accessor_helpers.rs]: Ooh, good call. I'll add that.
